fix(docs): Common 2025 news and docs (#36670)
truecharts/charts: website/src/content/docs/news/2025-06-26-Common-2025.md

---
slug: "news/common-2025"
title: "Common-2025 finally released"
authors: [alfi0812]
date: 2025-06-26
tags:
  - "2025"
---

## TrueCharts Releases `common-2025` Helm Library with Major Improvements

The TrueCharts team is proud to announce the release of **`common-2025`**, the latest version of the shared Helm chart library powering all TrueCharts charts. This major update delivers powerful new features, improved integrations, and essential structural upgrades that enhance usability, maintainability, and compatibility across the board.

## 🚀 Key Highlights of `common-2025`

- **🔐 Nginx Ingress Integration**
  - **Disabled by default** for a cleaner baseline
  - Built-in support for:
    - Authentication
    - IP whitelisting
    - ThemePark styling

- **🛠 GitOps CNPG Recovery Enhancements**

>   - No more unwanted Helm revision changes during CNPG cluster recovery

Reviewer: ```suggestion
  - No more need to bump revision numbers during CNPG cluster recovery
```

>   - Increasing the revision is now optional.

Reviewer: ```suggestion
  - Using the revision number is now optional.
```

- **🌐 Traefik Updates**
  - **Disabled by default** for a cleaner baseline
  - Complete rework of **Traefik middlewares**, now defined per chart

- **📦 Resource Management**
  - Updated defaults for **required pod resources**

- **🔑 Existing Pull Secret Support**
  - Added support for using **existing Kubernetes secrets** for container registry authentication
  - Simplifies integration in environments with pre-configured imagePullSecrets

- **🧩 Addon System Rework**

>   - Safer upgrades: prevents updates when legacy addon layouts are present

Reviewer: ```suggestion
  - Safe upgrade: prevents updates when some legacy addon layouts are present
```

  - **More configuration options for addons**, including:
    - Full access to the **workload definition**

- **📁 Structural Cleanups**
  - Removal of deprecated **Portal** (legacy from SCALE)
  - Major improvements to `notes.txt`

>   - Moved **MetalLB** and **Traefik service annotations** to centralized integration logic

Reviewer: ```suggestion
  - Moved **MetalLB** and **Traefik** service annotations to centralized integration logic
```

- **📊 Autoscaling Enhancements**

>   - Refactored **Horizontal Pod Autoscaler (HPA)**: dynamically scales replicas based on resource usage

Reviewer: ```suggestion
  - Refactored **Horizontal Pod Autoscaler (HPA)**: To comply to our coding standards and testing regime
```

>   - Added support for **Vertical Pod Autoscaler (VPA)**: automatically adjusts CPU and memory limits over time

Reviewer: ```suggestion
  - Added support for **Vertical Pod Autoscaler (VPA)**: Allows for automatically adjusts CPU and memory limits over time
```

- **👤 Service Account Overrides**
  - Added support to **optionally override the service account name** in the workload `podSpec`

- **🌐 Topology Key Configuration**
  - Users can now configure a **default `topologyKey`** for scheduling behavior

- **📌 Pod Affinity Support**

>   - Added full support for **pod affinity**, enabling more advanced workload placement strategies

Reviewer: ```suggestion
  - Added full support for **pod affinity**, enabling more advanced workload placement strategies
  - Default **pod affinity** to ensure pods sharing RWO PVCs are started on the same node
```

> 

Reviewer: ```suggestion
- **📊 Testing improvements**
  - Added more testing for less used features, such has HPA and for all newly introduced features

```
